pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/brokers/BrokerDataDownloader.kt

```diff
@@ -0,0 +1,131 @@
+        withContext(dispatcherProvider.io()) {
+            if (brokersToUpdate.isNotEmpty()) {
+                val extractFolder = File(context.filesDir, "unzipped-broker-json")
+                extractJsonFilesFromResponse(dbpService.getBrokerJsonFiles(), extractFolder)
+                processBrokerJsonFiles(extractFolder, brokersToUpdate)
+                deleteDirectory(extractFolder)
+            }
+        }
+    }
+
+    private fun extractJsonFilesFromResponse(
+        responseBody: ResponseBody,
+        outputDir: File,
+    ) {
+        Timber.d("PIR-update: Extracting data from $responseBody")
+        ZipInputStream(responseBody.byteStream()).use { zipInputStream ->
+            var entry = zipInputStream.nextEntry
+            while (entry != null) {
+                val outputFile = File(outputDir, entry.name)
+
+                if (entry.isDirectory) {
+                    outputFile.mkdirs()
+                } else {
+                    FileOutputStream(outputFile).use { outputStream ->
+                        zipInputStream.copyTo(outputStream)
+                    }
+                }
+
+                zipInputStream.closeEntry()
+                entry = zipInputStream.nextEntry
+            }
+        }
+        Timber.d("PIR-update: Extraction complete")
+    }
+
+    private suspend fun processBrokerJsonFiles(
+        directory: File,
+        brokersToUpdate: List<String>,
+    ) {
+        Timber.d("PIR-update: Attempting to process all relevant json for $brokersToUpdate")
+        if (!directory.exists() || !directory.isDirectory) {
+            Timber.e("PIR-update: $directory is not the json directory")
+            return
+        }
+
+        val adapter = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .add(StepsAsStringAdapter())
+            .build()
+            .adapter(PirJsonBroker::class.java)
+
+        directory.listFiles()?.get(0)?.listFiles { file ->
+            file.extension == "json" && brokersToUpdate.contains(file.name)
+        }?.forEach { jsonFile ->
+            Timber.d("PIR-update: Processing data from ${jsonFile.name}")
+            val content = jsonFile.readText() // Read JSON file as string
+            val broker = adapter.fromJson(content)
+            if (broker != null) {
+                pirRepository.updateBrokerData(jsonFile.name, broker)
+            }
+        }
+
+        Timber.d("PIR-update: Stored all new broker data")
+    }
+
+    private fun deleteDirectory(directory: File): Boolean {
+        if (directory.exists()) {
+            directory.listFiles()?.forEach { file ->
+                if (file.isDirectory) {
+                    deleteDirectory(file) // Recursively delete subdirectories
+                } else {
+                    file.delete() // Delete files
+                }
+            }
+        }
+        return directory.delete() // Finally, delete the empty folder
+    }
```

Comment:
I think okio already provides a method to delete directories recursively, something like this
```kotlin
SYSTEM.deleteRecursively("path/to/folder".toPath())
```